blis@0.9.0 %gcc fails on POWER: config_detect.c:(.text+0x3e8): undefined reference to `bli_cpuid_query_id': ./configure: line 1212: ./auto-detect.x: No such file or directory

### Steps to reproduce the issue

Trying to install `xyce@7.8.0 %gcc` on POWER, and the build is failing because dependency `blis@0.9.0` doesn't build.

This environment was building prior to https://github.com/spack/spack/pull/42194 and is now broken since the merge of https://github.com/spack/spack/commit/31ed39303cfb6039082c63f741781ce86cff97df

Spack environment:  [spack.yaml.txt](https://github.com/spack/spack/files/14058455/spack.yaml.txt)

<details><summary>Concretization</summary>

<pre>
==> Concretized xyce+mpi+pymi+pymi_static_tpls+shared
 -   tebguol  xyce@7.8.0%gcc@9.4.0~ipo+mpi~plugin+pymi+pymi_static_tpls+shared build_system=cmake build_type=Release cxxstd=11 generator=make patches=4d47cd1 arch=linux-ubuntu20.04-ppc64le
[+]  qmww7i3      ^bison@3.8.2%gcc@9.4.0 build_system=autotools arch=linux-ubuntu20.04-ppc64le
[+]  g6mxnw2          ^diffutils@3.9%gcc@9.4.0 build_system=autotools arch=linux-ubuntu20.04-ppc64le
[+]  ujjpuxo              ^libiconv@1.17%gcc@9.4.0 build_system=autotools libs=shared,static arch=linux-ubuntu20.04-ppc64le
[+]  44yev5y          ^gettext@0.22.4%gcc@9.4.0+bzip2+curses+git~libunistring+libxml2+pic+shared+tar+xz build_system=autotools arch=linux-ubuntu20.04-ppc64le
[+]  inr34es              ^libxml2@2.10.3%gcc@9.4.0+pic~python+shared build_system=autotools arch=linux-ubuntu20.04-ppc64le
[+]  nqsnqqs              ^tar@1.34%gcc@9.4.0 build_system=autotools zip=pigz arch=linux-ubuntu20.04-ppc64le
[+]  q6igweh                  ^pigz@2.8%gcc@9.4.0 build_system=makefile arch=linux-ubuntu20.04-ppc64le
[+]  zjiuoe3                  ^zstd@1.5.5%gcc@9.4.0+programs build_system=makefile compression=none libs=shared,static arch=linux-ubuntu20.04-ppc64le
[+]  wm4llbk          ^gnuconfig@2022-09-17%gcc@9.4.0 build_system=generic arch=linux-ubuntu20.04-ppc64le
[+]  5jv2xg5          ^m4@1.4.19%gcc@9.4.0+sigsegv build_system=autotools patches=9dc5fbd,bfdffa7 arch=linux-ubuntu20.04-ppc64le
[+]  kcqeonv              ^libsigsegv@2.14%gcc@9.4.0 build_system=autotools arch=linux-ubuntu20.04-ppc64le
[+]  eutfr7u          ^perl@5.38.0%gcc@9.4.0+cpanm+opcode+open+shared+threads build_system=generic patches=714e4d1 arch=linux-ubuntu20.04-ppc64le
[+]  h7q2zwi              ^berkeley-db@18.1.40%gcc@9.4.0+cxx~docs+stl build_system=autotools patches=26090f4,b231fcc arch=linux-ubuntu20.04-ppc64le
 -   dxp4enf      ^blis@0.9.0%gcc@9.4.0+blas+cblas build_system=makefile libs=shared,static threads=none arch=linux-ubuntu20.04-ppc64le
[+]  xnljdnz      ^cmake@3.27.9%gcc@9.4.0~doc+ncurses+ownlibs build_system=generic build_type=Release arch=linux-ubuntu20.04-ppc64le
[+]  wyqhhgj          ^curl@8.4.0%gcc@9.4.0~gssapi~ldap~libidn2~librtmp~libssh~libssh2+nghttp2 build_system=autotools libs=shared,static tls=openssl arch=linux-ubuntu20.04-ppc64le
[+]  q6tfp7y              ^nghttp2@1.57.0%gcc@9.4.0 build_system=autotools arch=linux-ubuntu20.04-ppc64le
[+]  7typ32i          ^ncurses@6.3%gcc@9.4.0~symlinks+termlib abi=none build_system=autotools arch=linux-ubuntu20.04-ppc64le
[+]  scdp26f          ^zlib-ng@2.1.5%gcc@9.4.0+compat+opt build_system=autotools arch=linux-ubuntu20.04-ppc64le
[+]  hwzu7ks      ^flex@2.6.3%gcc@9.4.0+lex~nls build_system=autotools arch=linux-ubuntu20.04-ppc64le
[+]  an6thpl          ^findutils@4.9.0%gcc@9.4.0 build_system=autotools patches=440b954 arch=linux-ubuntu20.04-ppc64le
[+]  ouuiatv      ^gcc-runtime@9.4.0%gcc@9.4.0 build_system=generic arch=linux-ubuntu20.04-ppc64le
[+]  xgvg43i      ^gmake@4.4.1%gcc@9.4.0~guile build_system=generic arch=linux-ubuntu20.04-ppc64le
[e]  723kgxt      ^mpich@4.1.2%gcc@9.4.0~argobots~cuda+fortran~hwloc+hydra+libxml2+pci~rocm+romio~slurm~vci~verbs~wrapperrpath build_system=autotools datatype-engine=auto device=ch4 netmod=ofi pmi=pmi arch=linux-ubuntu20.04-ppc64le
[+]  gfof2pb      ^openblas@0.3.26%gcc@9.4.0~bignuma~consistent_fpcsr+fortran~ilp64+locking+pic~shared build_system=makefile symbol_suffix=none threads=openmp arch=linux-ubuntu20.04-ppc64le
[+]  m5d7opt      ^py-pip@23.1.2%gcc@9.4.0 build_system=generic arch=linux-ubuntu20.04-ppc64le
[+]  ujx355i      ^py-pybind11@2.11.1%gcc@9.4.0~ipo build_system=cmake build_type=Release generator=ninja arch=linux-ubuntu20.04-ppc64le
[+]  3gvlxmd          ^ninja@1.11.1%gcc@9.4.0+re2c build_system=generic arch=linux-ubuntu20.04-ppc64le
[+]  5hlb2am              ^re2c@2.2%gcc@9.4.0 build_system=generic arch=linux-ubuntu20.04-ppc64le
[+]  n7a3p6b          ^py-setuptools@68.0.0%gcc@9.4.0 build_system=generic arch=linux-ubuntu20.04-ppc64le
[+]  tdzfu6r          ^py-wheel@0.41.2%gcc@9.4.0 build_system=generic arch=linux-ubuntu20.04-ppc64le
[+]  tv63vb3      ^python@3.11.6%gcc@9.4.0+bz2+crypt+ctypes+dbm~debug+libxml2+lzma~nis~optimizations+pic+pyexpat+pythoncmd+readline+shared+sqlite3+ssl~tkinter+uuid+zlib build_system=generic patches=13fa8bf,b0615b2,ebdca64,f2fd060 arch=linux-ubuntu20.04-ppc64le
[+]  gwzmqrp          ^bzip2@1.0.8%gcc@9.4.0~debug~pic+shared build_system=generic arch=linux-ubuntu20.04-ppc64le
[+]  6jjp63e          ^expat@2.5.0%gcc@9.4.0+libbsd build_system=autotools arch=linux-ubuntu20.04-ppc64le
[+]  lsirnds              ^libbsd@0.11.7%gcc@9.4.0 build_system=autotools arch=linux-ubuntu20.04-ppc64le
[+]  2p5ac5v                  ^libmd@1.0.4%gcc@9.4.0 build_system=autotools arch=linux-ubuntu20.04-ppc64le
[+]  ffzyvds          ^gdbm@1.23%gcc@9.4.0 build_system=autotools arch=linux-ubuntu20.04-ppc64le
[+]  k7ilxcw          ^libffi@3.4.4%gcc@9.4.0 build_system=autotools patches=070b1f3 arch=linux-ubuntu20.04-ppc64le
[+]  6322eig          ^libxcrypt@4.4.35%gcc@9.4.0~obsolete_api build_system=autotools patches=4885da3 arch=linux-ubuntu20.04-ppc64le
[+]  nshk57e          ^openssl@3.1.3%gcc@9.4.0~docs+shared build_system=generic certs=mozilla arch=linux-ubuntu20.04-ppc64le
[+]  zwlou7w              ^ca-certificates-mozilla@2023-05-30%gcc@9.4.0 build_system=generic arch=linux-ubuntu20.04-ppc64le
[+]  jz7tczu          ^pkgconf@1.9.5%gcc@9.4.0 build_system=autotools arch=linux-ubuntu20.04-ppc64le
[+]  vdxes6l          ^readline@8.2%gcc@9.4.0 build_system=autotools patches=bbf97f1 arch=linux-ubuntu20.04-ppc64le
[+]  hkdnngo          ^sqlite@3.43.2%gcc@9.4.0+column_metadata+dynamic_extensions+fts~functions+rtree build_system=autotools arch=linux-ubuntu20.04-ppc64le
[+]  gb7n4ic          ^util-linux-uuid@2.38.1%gcc@9.4.0 build_system=autotools arch=linux-ubuntu20.04-ppc64le
[+]  x2hkpf3          ^xz@5.4.1%gcc@9.4.0+pic build_system=autotools libs=shared,static arch=linux-ubuntu20.04-ppc64le
[+]  lz45rvn      ^trilinos@15.0.0%gcc@9.4.0~adelus~adios2+amesos+amesos2~anasazi+aztec+basker+belos+boost~chaco+complex~cuda~cuda_rdc~debug~dtk+epetra+epetraext+epetraextbtf+epetraextexperimental+epetraextgraphreorderings~exodus+explicit_template_instantiation~float+fortran~gtest~hdf5~hypre+ifpack~ifpack2+intrepid+intrepid2~ipo+isorropia+kokkos~mesquite+minitensor~ml+mpi~muelu~mumps+nox~openmp~panzer+phalanx+piro~python~rocm~rocm_rdc+rol+rythmos+sacado~scorec+shards+shared+shylu+stk+stokhos+stratimikos~strumpack+suite-sparse~superlu~superlu-dist~teko+tempus~test+thyra+tpetra+trilinoscouplings~wrapper~x11+zoltan~zoltan2 build_system=cmake build_type=Release cxxstd=17 generator=make gotype=long_long arch=linux-ubuntu20.04-ppc64le
[+]  rl2lzpy          ^boost@1.84.0%gcc@9.4.0~atomic~chrono~clanglibcpp~container~context~contract~coroutine~date_time~debug+exception~fiber~filesystem+graph~graph_parallel~icu~iostreams~json~locale~log+math+mpi+multithreaded~nowide~numpy~pic~program_options~python~random~regex~serialization+shared~signals~singlethreaded+stacktrace~system~taggedlayout~test~thread~timer~type_erasure~versionedlayout~wave build_system=generic cxxstd=11 patches=a440f96 visibility=hidden arch=linux-ubuntu20.04-ppc64le
[+]  gzsndre          ^hwloc@2.9.1%gcc@9.4.0~cairo~cuda~gl~libudev+libxml2~netloc~nvml~oneapi-level-zero~opencl+pci~rocm build_system=autotools libs=shared,static arch=linux-ubuntu20.04-ppc64le
[+]  kykfti7              ^libpciaccess@0.17%gcc@9.4.0 build_system=autotools arch=linux-ubuntu20.04-ppc64le
[+]  mlqafjp                  ^libtool@2.4.7%gcc@9.4.0 build_system=autotools arch=linux-ubuntu20.04-ppc64le
[+]  bxz4xir                  ^util-macros@1.19.3%gcc@9.4.0 build_system=autotools arch=linux-ubuntu20.04-ppc64le
[+]  4qq7ts4          ^kokkos@4.1.00%gcc@9.4.0~aggressive_vectorization~compiler_warnings~cuda~debug~debug_bounds_check~debug_dualview_modify_check~deprecated_code~examples~hpx~hpx_async_dispatch~hwloc~ipo~memkind~numactl~openmp~openmptarget~pic~rocm+serial+shared~sycl~tests~threads~tuning~wrapper build_system=cmake build_type=Release cxxstd=17 generator=make intel_gpu_arch=none arch=linux-ubuntu20.04-ppc64le
[+]  gi22jpx          ^metis@5.1.0%gcc@9.4.0~gdb~int64~ipo~real64+shared build_system=cmake build_type=Release generator=make patches=4991da9,93a7903,b1225da arch=linux-ubuntu20.04-ppc64le
[+]  3l2veem          ^parmetis@4.0.3%gcc@9.4.0~gdb~int64~ipo+shared build_system=cmake build_type=Release generator=make patches=4f89253,50ed208,704b84f arch=linux-ubuntu20.04-ppc64le
[+]  cghugnd          ^suite-sparse@5.13.0%gcc@9.4.0~cuda~graphblas~openmp+pic build_system=generic arch=linux-ubuntu20.04-ppc64le
[+]  mprdukh              ^gmp@6.2.1%gcc@9.4.0+cxx build_system=autotools libs=shared,static patches=69ad2e2 arch=linux-ubuntu20.04-ppc64le
[+]  eymje5n                  ^autoconf@2.72%gcc@9.4.0 build_system=autotools arch=linux-ubuntu20.04-ppc64le
[+]  vv2ynja                  ^automake@1.16.5%gcc@9.4.0 build_system=autotools arch=linux-ubuntu20.04-ppc64le
[+]  5ndoami              ^mpfr@4.2.0%gcc@9.4.0 build_system=autotools libs=shared,static arch=linux-ubuntu20.04-ppc64le
[+]  gfsk7zk                  ^autoconf-archive@2023.02.20%gcc@9.4.0 build_system=autotools arch=linux-ubuntu20.04-ppc64le
[+]  enfke5e                  ^texinfo@7.0.3%gcc@9.4.0 build_system=autotools arch=linux-ubuntu20.04-ppc64le
</pre></details>

Building spack.yaml posted above in Docker container image `ecpe4s/ubuntu20.04-runner-ppc64-gcc-11.4-mpi-base:2023.08.20` produces this error:
```
$> git clone https://github.com/spack/spack
$> git -C /spack checkout 31ed39303cfb6039082c63f741781ce86cff97df
$> . spack/share/spack/setup-env.sh
$> spack mirror add E4S https://cache.e4s.io/24.02
$> spack buildcache keys -it

$> spack -e . install
...
==> Installing blis-0.9.0-dxp4enfutyglpo46e7keaqumdil4bgdj [45/46]
==> No binary for blis-0.9.0-dxp4enfutyglpo46e7keaqumdil4bgdj found: installing from source
==> Fetching https://mirror.spack.io/_source-cache/archive/11/1135f664be7355427b91025075562805cdc6cc730d3173f83533b2c5dcc2f308.tar.gz
==> No patches needed for blis
==> blis: Executing phase: 'edit'
==> Error: ProcessError: Command exited with status 1:
    './configure' '--prefix=/spack/opt/spack/linux-ubuntu20.04-ppc64le/gcc-9.4.0/blis-0.9.0-dxp4enfutyglpo46e7keaqumdil4bgdj' '--enable-threading=none' '--enable-cblas' '--enable-blas' '--enable-shared' '--enable-static' 'auto'

3 errors found in build log:
     45    configure: configuring with official version string.
     46    configure: found shared library .so version '4.0.0'.
     47    configure:   .so major version: 4
     48    configure:   .so minor.build version: 0.0
     49    configure: automatic configuration requested.
     50    /usr/bin/ld: /tmp/ccIh3IX8.o: in function `main':
  >> 51    config_detect.c:(.text+0x3e8): undefined reference to `bli_cpuid_query_id'
  >> 52    collect2: error: ld returned 1 exit status
  >> 53    ./configure: line 1212: ./auto-detect.x: No such file or directory
     54    configure: hardware detection driver returned ''.
     55    configure: checking configuration against contents of 'config_registry'.
     56    configure: 'auto-detected configuration '' is NOT registered!
     57    configure:
     58    configure: *** Cannot continue with unregistered configuration ''. ***
     59    configure:

See build log for details:
  /tmp/root/spack-stage/spack-stage-blis-0.9.0-dxp4enfutyglpo46e7keaqumdil4bgdj/spack-build-out.txt
```

### Error message

<details><summary>Error message</summary>

<pre>
... see above
</pre></details>


### Information on your system

* **Spack:** 0.22.0.dev0 (31ed39303cfb6039082c63f741781ce86cff97df)
* **Python:** 3.8.10
* **Platform:** linux-ubuntu20.04-power9le
* **Concretizer:** clingo

### Additional information

[spack-build-env.txt](https://github.com/spack/spack/files/14058472/spack-build-env.txt)
[spack-build-out.txt](https://github.com/spack/spack/files/14058473/spack-build-out.txt)

@kuberry @tbird2001

### General information

- [X] I have run `spack debug report` and reported the version of Spack/Python/Platform
- [X] I have run `spack maintainers <name-of-the-package>` and **@mentioned** any maintainers
- [X] I have uploaded the build log and environment files
- [X] I have searched the issues of this repo and believe this is not a duplicate

--- a/var/spack/repos/builtin/packages/xyce/package.py
+++ b/var/spack/repos/builtin/packages/xyce/package.py
@@ -138,8 +138,8 @@ class Xyce(CMakePackage):
 
         depends_on("armpl-gcc~shared", when="^armpl-gcc")
         depends_on("atlas~shared", when="^atlas")
-        depends_on("blis libs=static", when="^blis+cblas")
-        depends_on("blis libs=static", when="^blis+blas")
+        depends_on("blis libs=static", when="^[virtuals=blas] blis+cblas")
+        depends_on("blis libs=static", when="^[virtuals=blas] blis+blas")
         depends_on("clblast~shared", when="^clblast+netlib")
         depends_on("intel-mkl~shared", when="^intel-mkl")
         depends_on("intel-oneapi-mkl~shared", when="^intel-oneapi-mkl")

